### PR TITLE
Remove overlapping outputs detecting init script

### PR DESCRIPTION
**Pull request by lptr**
It is not a recommended way to track overlapping outputs anymore.

### CODE REVIEW

File: subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/CacheInitScriptsSmokeTest.groovy [hunk 1]
```diff
@@ -22,27 +22,18 @@ class CacheInitScriptsSmokeTest extends AbstractSmokeTest {
 
     def 'cache init script plain text output'() {
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/CacheInitScriptsSmokeTest.groovy]: Looks like we can remove the whole smoke test now, given that is was only used to test the cache init scripts - which this PR removes.

Comment (lptr) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/CacheInitScriptsSmokeTest.groovy]: It still tests that the cache can be initialized from an init script. Do you want to remove that as well?

Comment (wolfs) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/CacheInitScriptsSmokeTest.groovy]: Let's remove it, we already have a test for that: https://github.com/gradle/gradle/blob/06677b922432b742db3b81fb3938616f83b82387/subprojects/core/src/integTest/groovy/org/gradle/caching/configuration/internal/BuildCacheConfigurationIntegrationTest.groovy#L79-L96

Testing that the cache can be initialized from an init script should not be a smoke test.

Comment (lptr) [reply subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/CacheInitScriptsSmokeTest.groovy]: Removed.

File: subprojects/smoke-test/src/smokeTest/resources/org/gradle/smoketests/cache-init-scripts/build.gradle [hunk 2]
```diff
@@ -1,3 +1 @@
-apply from: "overlapping-task-outputs-stats-init.gradle"
```
Comment (wolfs) [inline subprojects/smoke-test/src/smokeTest/resources/org/gradle/smoketests/cache-init-scripts/build.gradle]: This file can go away completely as well.

Comment (lptr) [reply subprojects/smoke-test/src/smokeTest/resources/org/gradle/smoketests/cache-init-scripts/build.gradle]: Ah, you're right. Actually, the whole directory can come out.

**Pull request merged**